specification/azureactivedirectory/resource-manager/Microsoft.Aadiam/stable/2017-04-01/azureactivedirectory.json

```diff
@@ -0,0 +1,507 @@
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "Successful request to get more information about diagnostic setting",
+            "schema": {
+              "$ref": "#/definitions/DiagnosticSettingsResource"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Gets the diagnostic setting": {
+            "$ref": "./examples/getDiagnosticSetting.json"
+          }
+        } 
+      },
+      "put": {
+        "tags": ["DiagnosticSettings"],
+        "operationId": "DiagnosticSettings_CreateOrUpdate",
+        "x-ms-examples": {
+          "BatchAccountDelete": { "$ref": "./examples/createOrUpdateDiagnosticSetting.json" }
+        },
+        "description": "Creates or updates diagnostic settings for AadIam.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/DiagnosticSettingsResource"
+            },
+            "description": "Parameters supplied to the operation."
+          },
+          {
+            "$ref": "#/parameters/NameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful request to create AadIam diagnostic setting.",
+            "schema": {
+              "$ref": "#/definitions/DiagnosticSettingsResource"
+            }
+          },
+          "default": {
+            "description": "Error. The request to create diagnostic setting has failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "delete": {
+        "tags": ["DiagnosticSettings"],
+        "operationId": "DiagnosticSettings_Delete",
+        "description": "Deletes existing diagnostic setting for AadIam.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/NameParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error. The request to delete named diagnostic setting has failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "Successful request to remove a diagnostic setting"
+          },
+          "204": {
+            "description": "Successful request to remove a diagnostic setting"
+          }
+        },
+        "x-ms-examples": {
+          "Deletes the diagnostic setting": {
+            "$ref": "./examples/deleteDiagnosticSetting.json"
+          }
+        } 
+      }
+    },
+    "/providers/microsoft.aadiam/diagnosticSettingsCategories": {
+      "get": {
```

Comment:
I suggest adding x-ms-paggeable extension for listing: http://azure.github.io/autorest/extensions/#x-ms-pageable 
The Canonical way to use this extension is 
``` json
"x-ms-pageable":  {                         
"nextLinkName": "nextLink"                  
}
```
inside the Operation Object.